tests/integration_testing.go

```diff
@@ -0,0 +1,157 @@
+package integration_testing
+
+import (
+        "bytes"
+        "fmt"
+        "os"
+        "os/exec"
+        "runtime"
+        "strings"
+        "testing"
+        "github.com/bazelbuild/rules_go/go/tools/bazel"
+        "github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M, workspace string) {
+        if err := ParseArgs(); err != nil {
+                fmt.Fprint(os.Stderr, err)
+                return
+        }
+	defer exec.Command(Context.BazelBinary, "shutdown")
+
+        bazel_testing.TestMain(m, bazel_testing.Args{
+                Main: workspace + GenerateBazelrc(),
+        })
+}
+
+func AssertOutput(t *testing.T, output []byte, expected string) {
+        if string(output) != expected {
+                t.Fatalf("output of bazel process is invalid.\nExpected: %v\n, Actual: %v\n", expected, string(output))
+        }
+}
+
+var Context struct {
+        Nixpkgs bool
+        BazelBinary string
+}
+
+func ParseArgs() error {
+        bazelPath := ""
+        for _, arg := range os.Args {
+                if strings.HasPrefix(arg, "nixpkgs=") {
+                        fmt.Sscanf(arg, "nixpkgs=%t", &Context.Nixpkgs)
+                } else if strings.HasPrefix(arg, "bazel_bin=") {
+                        fmt.Sscanf(arg, "bazel_bin=%s", &bazelPath)
+                }
+        }
+        bazelAbsPath, err := bazel.Runfile(bazelPath)
+        Context.BazelBinary = bazelAbsPath
+        return err
+}
+
+func GenerateBazelrc() string {
+        bazelrc := `
+-- .bazelrc --
+build:linux-nixpkgs --host_platform=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host
+build:linux-nixpkgs --incompatible_enable_cc_toolchain_resolution
+build:macos-nixpkgs --host_platform=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host
+build:macos-nixpkgs --incompatible_enable_cc_toolchain_resolution
+build:linux-bindist --incompatible_enable_cc_toolchain_resolution
+build:macos-bindist --incompatible_enable_cc_toolchain_resolution
+build:windows-bindist --crosstool_top=@rules_haskell_ghc_windows_amd64//:cc_toolchain
+`
+        if bazel_testing.OutputUserRoot != "" {
+                bazelrc += fmt.Sprintf("startup: --output_user_root=%s\n", bazel_testing.OutputUserRoot)
+        }
+        return bazelrc
+}
+
+func BazelEnv() []string {
+        env := []string{}
+	// It's important value of $HOME to be invariant between different integration test runs
+        // and to be writable directory for bazel test.
```

Comment:
```suggestion
	// It's important that the value of $HOME is invariant between different integration test runs
        // and that the directory is writable for bazel test.
```
Maybe a bit easier to read?

Reply:
Fixed, sorry for my runglish :)